File: cinebook-frontend/src/services/bookingService.js
import axiosInstance from "./axiosinstance";

export const createBooking=(bookingdata)=>{
    return axiosInstance.post("/api/booking",bookingdata);
}
export const getMyBookings=()=>{
      return axiosInstance.get("/api/booking/my-bookings");
}

export const cancelBooking=(bookingId)=>{
    return axiosInstance.put(`/api/booking/cancel/${bookingId}`);
}

export const getBookingById=(bookingId)=>{
    return axiosInstance.get(`/api/booking/${bookingId}`);
}

export const searchBooking=(bookingNumber)=>{
    return axiosInstance.get(`/api/booking/search/${bookingNumber}`);
}
export const searchByBookingStatus=(bookingStatus)=>{
    return axiosInstance.get(`/api/booking/status/${bookingStatus}`);
}
export const allBookings=()=>{
    return axiosInstance.get("/api/booking/all");
}

File: cinebook-backend/cinebook-backend/src/main/java/com/cinebook/cinebook/service/BookingService.java
package com.cinebook.cinebook.service;

import com.cinebook.cinebook.dto.request.BookingRequestDto;
import com.cinebook.cinebook.dto.response.BookingResponseDto;
import com.cinebook.cinebook.dto.response.MovieResponseDto;
import com.cinebook.cinebook.dto.response.MovieRevenueDto;
import com.cinebook.cinebook.entity.Booking;
import com.cinebook.cinebook.entity.BookingSeat;
import com.cinebook.cinebook.enums.BookingStatus;
import com.cinebook.cinebook.enums.SeatType;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;


public interface BookingService {
    BookingResponseDto createBooking(BookingRequestDto request);
   Page<BookingResponseDto> getAllBookings(int page,int size,String sortBy,String sortDir);
    BookingResponseDto getBookingById(Long bookingId);
      List<BookingResponseDto>searchByBookingNumber(String BookingNumber);
      List<BookingResponseDto>searchByStatus(BookingStatus bookingStatus);

      List<BookingResponseDto> getMyBookings();
      void cancelBooking(Long bookingId);


}
